File: mobile/src/main/java/org/horizontal/tella/mobile/domain/repository/IUWAZIServersRepository.java
package org.horizontal.tella.mobile.domain.repository;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Single;
import org.horizontal.tella.mobile.domain.entity.UWaziUploadServer;

public interface IUWAZIServersRepository {
    Single<List<UWaziUploadServer>> listUwaziServers();
    Single<UWaziUploadServer> createUWAZIServer(UWaziUploadServer server);
    Completable removeUwaziServer(long id);
    Single<UWaziUploadServer> updateUwaziServer(UWaziUploadServer server);
    Single<Long> countUwaziServers();
    Single<UWaziUploadServer> getUwaziServerById(Long serverID);
}

File: mobile/src/main/java/org/horizontal/tella/mobile/domain/repository/uwazi/IUWAZIServersRepository.kt
package org.horizontal.tella.mobile.domain.repository.uwazi

import org.horizontal.tella.mobile.domain.entity.UWaziUploadServer

interface IUWAZIServersRepository {
    suspend fun listUwaziServers(): List<UWaziUploadServer>
    suspend fun createUWAZIServer(server: UWaziUploadServer): UWaziUploadServer?
    suspend fun removeUwaziServer(id: Long)
    suspend fun updateUwaziServer(server: UWaziUploadServer) : UWaziUploadServer?
    suspend fun countUwaziServers(): Long?
}